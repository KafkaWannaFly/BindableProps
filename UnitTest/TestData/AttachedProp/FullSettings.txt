namespace MyMauiApp.Controls
{
    public partial class MyMauiControl : ContentView
    {
        [AttachedProp(
            DefaultBindingMode = ((int)BindingMode.OneWay),
            ValidateValueDelegate = nameof(ValidateValue),
            PropertyChangedDelegate = nameof(PropertyChangedDelegate),
            PropertyChangingDelegate = nameof(PropertyChangingDelegate),
            CoerceValueDelegate = nameof(CoerceValueDelegate),
            CreateDefaultValueDelegate = nameof(CreateDefaultValueDelegate)
        )]
        private bool _hasShadow = false;
    
        public MyMauiControl()
        {
            // My logic
        }
        
        static bool ValidateValue(BindableObject bindable, object value)
        {
            return true;
        }
    
        static void PropertyChangedDelegate(BindableObject bindable, object oldValue, object newValue)
        {
            // Do something
        }
    
        static void PropertyChangingDelegate(BindableObject bindable, object oldValue, object newValue)
        {
            // Do something
        }
    
        static object CoerceValueDelegate(BindableObject bindable, object value)
        {
            // Do something
            return 0;
        }
    
        static object CreateDefaultValueDelegate(BindableObject bindable)
        {
            // Do something
            return string.Empty;
        }
    }
    
}