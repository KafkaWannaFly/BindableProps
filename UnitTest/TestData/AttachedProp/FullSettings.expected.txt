#pragma warning disable

// <auto-generated/>

namespace MyMauiApp.Controls
{
    public partial class MyMauiControl
    {
        public static readonly BindableProperty HasShadowProperty = BindableProperty.CreateAttached(
            "HasShadow",
            typeof(bool),
            typeof(MyMauiControl),
            false,
            (BindingMode)((int)BindingMode.OneWay),
            ValidateValue,
            PropertyChangedDelegate,
            PropertyChangingDelegate,
            CoerceValueDelegate,
            CreateDefaultValueDelegate
        );
        
        public static bool GetHasShadow(BindableObject view)
        {
            return (bool)view.GetValue(HasShadowProperty);
        }

        public static void SetHasShadow(BindableObject view, bool value)
        {
            view.SetValue(HasShadowProperty, value);
        }
    }
}